### PR TITLE
Custom SmartRouter image

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

**Comment by tchughesiv** [comment]
lgtm

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,139 @@
+[id='customimage-smartrouter-proc_{context}']
+:componentimage: smartrouter
+:component: Smart Router
+= Creating a custom Smart Router image with an additional JAR file to implement custom routing
+
+By default, Smart Router routes requests based on the container alias. If severak {KIE_SERVERS} provide a servcie with teh same container alias, Smart Router balances the load between them.
```
Comment (spolti) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:6]: typo, severak

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+. Build the source code with Maven:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mvn clean package
+----
++
+The build process generates the following JAR file: `target/router-ext-0.0.1-SNAPSHOT.jar` 
++
+. Create a work directory for creating the custom image, copy the generated JAR file into the directory, and then change to the directory, for example:
```
Comment (spolti) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:30]: Create a working directory sounds better, wdyt?

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mkdir /tmp/smartrouter
+cp target/router-ext-0.0.1-SNAPSHOT.jar /tmp/smartrouter
+cd /tmp/smartrouter
+----
++
+include::customimage-download-step.adoc[]
+. To extract the `openshift-launch.sh` file from the official Smart Router image, enter the following command:
```
Comment (spolti) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:40]: MAybe add more context here. like

To customize the Smart ROuter image to use the custom routing .... you'll need to extract the 'openshift-launch.sh' file from the official smart router image..

**Review by spolti — CHANGES_REQUESTED** [review]

**Review by spolti — APPROVED** [review]

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+.Prerequisites
+
+* A JDK and Apache Maven are installed.
+* The project for deploying {PRODUCT} is created in your {OPENSHIFT} environment
+* You know the route for the {OPENSHIFT} image registry and have the permission to push images into the registry. For instructions about configuring the registry, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/registry/index[_Registry_] in {OPENSHIFT} product documentation.
+
+.Procedure
+
+. Download the `{PRODUCT_FILE}-container-sources.zip` product deliverable file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
```
Comment (michelehaglund) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:18]: I just addressed a DDF ticket about locating the zip file in the Sofware Downloads page on the Portal. I just clicked on all of the names in https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=rhpam&productChanged=yes and none of them were for the rhpam-7.9.0-container-sources.zip file.

So, I would confirm the link name and reword the sentence as in the following example: Note that I just used an example link. It is not the correct one.

Click Red Hat Process Automation Manager 7.8.0 Process Server for All Supported EE7 Containers to download the rhpam-7.9.0-container-sources.zip product deliverable file from the Software Downloads page of the Red Hat Customer Portal.

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+. Extract the contents of the file and change to the `/src/rhpam-7-openshift-image/quickstarts/router-ext` directory.
+. Modify the sample source of the router extension in this directory as necessary. The existing code implements simple routing based on the version of the container.
```
Comment (michelehaglund) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:20]: "this directory"? Which directory?

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+[subs="attributes,verbatim,macros,quotes"]
+----
+exec ${JAVA_HOME}/bin/java ${SHOW_JVM_SETTINGS} "${D_ARR[@]}" \
+    -cp /opt/${JBOSS_PRODUCT}/router-ext-0.0.1-SNAPSHOT.jar:/opt/${JBOSS_PRODUCT}/${KIE_ROUTER_DISTRIBUTION_JAR} \
+    org.kie.server.router.KieServerRouter
+----
++
+This change adds the extension JAR file to the Java Class Path.
++
+. Create a `Dockerfile` file that defines a custom image based on the base image. The file must add the JAR file and the new `openshift-launch.sh` file, change the current user to `root`, set the necessary permissions for the `openshift-launch.sh` file, and then revert to `USER 185`, the {EAP} user. The following example shows the content of the `Dockerfile` file:
```
Comment (michelehaglund) [inline doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:66]: Minor rewrite suggestion:

Create a `Dockerfile` file that defines a custom image based on the base image. The file must add the JAR file and the new `openshift-launch.sh` file. Change the current user to `root`, set the necessary permissions for the `openshift-launch.sh` file, and then revert to `USER 185`, which is the {EAP} user. The following example shows the content of the `Dockerfile` file:

**Review by michelehaglund — APPROVED** [review]
Minor comments. Looks good @mramendi

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc (hunk 4 above)
Comment (mramendi) [reply doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:18]: I will review my links when CR1 is out, thanks! For now keeping for consistency with other docs

File: doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc (hunk 6 above)
Comment (mramendi) [reply doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc:66]: The file must do all this - I'll try to make it clearer.

**Pull request merged**